.actions/assistant.py

```diff
@@ -168,6 +185,49 @@ def mirror_pkg2source(pypi_folder: str, src_folder: str) -> None:
                 shutil.rmtree(py_dir2, ignore_errors=True)
                 shutil.copytree(py_dir, py_dir2)
 
+    @staticmethod
+    def copy_replace_imports(
+        source_dir: str, source_import: str, target_import: str, target_dir: Optional[str] = None
+    ) -> None:
+        """Recursively replace imports in given folder."""
+
+        source_imports = source_import.strip().split(",")
+        target_imports = target_import.strip().split(",")
+        assert len(source_imports) == len(target_imports), (
+            "source and target imports must have the same length, "
+            f"source: {len(source_import)}, target: {len(target_import)}"
+        )
+
+        if target_dir is None:
+            target_dir = source_dir
+
+        ls = _retrieve_files(source_dir)
+
+        for fp in ls:
+            if not fp.endswith(".py"):
+                if not fp.endswith(".pyc"):
+                    fp_new = fp.replace(source_dir, target_dir)
```

Comment:
what is this block for? markdown files and such?

Reply:
Exactly. Basically anything that isn't python. JS, MD, images, etc

Reply:
Okay. I would say this is simpler to read then


```python
            if fp.endswith(".py"):
                with open(fp, encoding="utf-8") as fo:
                    py = fo.readlines()

                for _source_import, _target_import in zip(source_imports, target_imports):
                    for i, ln in enumerate(py):
                        py[i] = re.sub(rf"(?!_){_source_import}(?!_)", _target_import, ln)

                if target_dir:
                    fp_new = fp.replace(source_dir, target_dir)
                    os.makedirs(os.path.dirname(fp_new), exist_ok=True)
                else:
                    fp_new = fp

                with open(fp_new, "w", encoding="utf-8") as fo:
                    fo.writelines(py)
            elif not fp.endswith("pyc") :
                fp_new = fp.replace(source_dir, target_dir)
                os.makedirs(os.path.dirname(fp_new), exist_ok=True)
                if os.path.abspath(fp) != os.path.abspath(fp_new):
                    shutil.copy2(fp, fp_new)
```

Reply:
pls send it as suggestion :)